compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -1397,19 +1397,30 @@ class JSCodeGen()(implicit ctx: Context) {
     } else /*if (isScalaJSDefinedJSClass(currentClassSym)) {
       genJSSuperCall(tree, isStat)
     } else*/ {
-      val superCall = genApplyMethodStatically(
-          genThis()(sup.span), sym, genActualArgs(sym, args))
-
-      // Initialize the module instance just after the super constructor call.
-      if (isStaticModule(currentClassSym) && !isModuleInitialized &&
-          currentMethodSym.get.isClassConstructor) {
-        isModuleInitialized = true
-        val className = encodeClassName(currentClassSym)
-        val thisType = jstpe.ClassType(className)
-        val initModule = js.StoreModule(className, js.This()(thisType))
-        js.Block(superCall, initModule)
+      if (sym.owner.isAllOf(Scala2x | Trait) && sym.isConstructor) {
+        // calling genApplyMethodStatically will genereate a method with one argument for `this`, so we do it manually a zero-argument method
+        assert(args.size == 0)
+        val sym0 = sym.owner.info.decl(nme.TRAIT_CONSTRUCTOR).symbol.name.mangledString
+        val methodIdent = js.MethodIdent(MethodName(SimpleMethodName(sym0), List.empty, jstpe.VoidRef, false))
```

Comment:
`sym0` is always going to be the string `"$init$"`. It's pointless to go through all those hoops to compute that constant string.

Since it's constant, the `MethodName` will always be the same here, and can be pre-computed in a `val` inside the companion object of `JSCodeGen`, like:
```scala
object JSCodeGen {
  ...

  private val TraitInitMethodName = MethodName("$init$", Nil, jstpe.VoidRef)

  ...
}
```